src/k8s/pkg/snap/snap.go

```diff
@@ -159,10 +159,18 @@ func (s *snap) KubernetesPKIDir() string {
 	return "/etc/kubernetes/pki"
 }
 
+func (s *snap) EtcdPKIDir() string {
+	return "/etc/kubernetes/pki/etcd"
+}
+
 func (s *snap) KubeletRootDir() string {
 	return "/var/lib/kubelet"
 }
 
+func (s *snap) SupportedDatastores() []string {
+	return []string{"k8s-dqlite", "external"}
```

Comment:
Could this also just be a `const/var SupportedDatastores` in `setup`? I don't see why it needs to be part of the snap interface.